### PR TITLE
Reformat code according to EditorConfig rules

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

Reformat source code with [dotnet/format](https://github.com/dotnet/format) according to rules in the `.editorconfig` file.
```
dotnet tool install --global dotnet-format --version 3.1.37601
dotnet-format
```

The following rules were disabled as they resulted in a high number of changes:
```
csharp_preserve_single_line_statements = false
csharp_new_line_before_open_brace = all
```

To exclude regressions, we could create a new static test:
```
# Terminates with a non-zero exit code if any files were formatted
dotnet-format --check --dry-run
```
## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Review by iSazonov — COMMENTED** [review]
@xtqqczze Please revert changes in .editorconfig. They was needed only locally to allow you to format but we should keep these options as they correspond to our code convention.
(For static test we could use another config file with fewer rules.)

> Which rules would you like to see added?

I see many rules in .Net Core and Roslyn .editorconfig-s and we could review them and add some ones to our config. When I added these rules last time I had to remove many because there were a lot of problems. Now the code is cleaner and we can add more rules.

Also please see CodeFactor issues. We need to adjust CodeFactor or formatting options to resolve them.

**Comment by xtqqczze** [comment]
CodeFactor issues are violations of [SA1008](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1008.md) due to added space after the `delegate` keyword. Rule documentation states an opening parenthesis can be preceded by whitespace when it is preceded by "certain C# keywords" but is ambiguous as to whether or not this applies to the delegate keyword. There is an open issue https://github.com/dotnet/roslyn/issues/3257 concerning this behaviour.

We can apply a refactoring to replace the anonymous methods with `delegate` keyword with lamba expressions and avoid the rule violations. This would be a workaround but is also a recommended refactoring.

**Comment by xtqqczze** [comment]
@iSazonov This PR disables those rules because it is apparent the code is not already in the prescribed style. We could open a new issue to discuss whether we want these rules in our code convention?

**Comment by xtqqczze** [comment]
rebased to restart CI

**Comment by iSazonov** [comment]
> This PR disables those rules because it is apparent the code is not already in the prescribed style. 

We have over 50000 CodeFactor issues but we do not disable the rules because we don't want add _new_ issues.

> We could open a new issue to discuss which rules we want for our code convention?

We already have Code Conventions. This allows us add any rules to suggest/force the Code Conventions.

**Comment by iSazonov** [comment]
@xtqqczze Please resolve merge conflict.

**Comment by xtqqczze** [comment]
@iSazonov rebased and removed commit "Remove duplicate semicolons" This should be part of a seperate PR as the change is unrelated to using the `dotnet-format` tool.

**Comment by xtqqczze** [comment]
@iSazonov We should make changes to rules in .editorconfig before creating a new dotnet-format PR.

**Review by iSazonov — APPROVED** [review]

**Comment by TravisEz13** [comment]
Please resolve merge conflicts.
Such a large PR may be difficult to get merged.

**Comment by xtqqczze** [comment]
@TravisEz13 rebased

**Comment by xtqqczze** [comment]
@TravisEz13 rebased, again

File: src/System.Management.Automation/engine/debugger/debugger.cs [hunk 1]
```diff
@@ -644,8 +644,8 @@ public virtual IEnumerable<CallStackFrame> GetCallStack()
         /// Adds the provided set of breakpoints to the debugger.
         /// </summary>
         /// <param name="breakpoints">Breakpoints.</param>
-        /// <param name="runspaceId">The runspace id of the runspace you want to interact with, null being the current runspace.</param>
-        public virtual void SetBreakpoints(IEnumerable<Breakpoint> breakpoints, int? runspaceId) =>  
+        /// <param name="runspaceId">The runspace id of the runspace you want to interact with. Defaults to null (current runspace).</param>
+        public virtual void SetBreakpoints(IEnumerable<Breakpoint> breakpoints, int? runspaceId = null) =>
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/debugger/debugger.cs]: It is not formatting change - please revert.

Comment (xtqqczze) [reply src/System.Management.Automation/engine/debugger/debugger.cs]: Good catch, this looks like incorrect merge

Comment (xtqqczze) [reply src/System.Management.Automation/engine/debugger/debugger.cs]: Yes, this was from incorrect merge, I will rewrite commit.

**Comment by xtqqczze** [comment]
`PowerShell-CI-static-analysis` failures are in markdown-links and are unrelated.

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**